src/sync.h

```diff
@@ -318,4 +319,22 @@ struct SCOPED_LOCKABLE LockAssertion
     ~LockAssertion() UNLOCK_FUNCTION() {}
 };
 
+class CLockFreeGuard
+{
+    std::atomic_bool& lock;
+public:
+    CLockFreeGuard(std::atomic_bool& lock) : lock(lock)
+    {
+        bool desired = false;
+        while (!lock.compare_exchange_weak(desired, true,
+                                           std::memory_order_release,
+                                           std::memory_order_relaxed))
+            desired = false;
+    }
```

Comment:
This loop will consume a lot of CPU resources.

Reply:
Force pushed a fix.